### PR TITLE
Add Ignored Packages

### PR DESCRIPTION
**Pull request by Leapfrognz**
Adds the ability to define packages to ignore via yml

### CODE REVIEW

File: src/Extensions/CheckComposerUpdatesExtension.php [hunk 1]
```diff
@@ -38,6 +38,7 @@ public function updatePackageInfo(array &$installedPackageList)
         // Fetch types of packages that are "allowed" - ie. dependencies that we actually care about
         $allowedTypes = (array) Config::inst()->get(UpdatePackageInfoTask::class, 'allowed_types');
         $composerPackagesAndConstraints = $this->owner->getComposerLoader()->getPackages($allowedTypes);
+        $ignoredPackages = (array) Config::inst()->get(self::class, 'ignored_packages');
```
Comment (robbieaverill) [inline src/Extensions/CheckComposerUpdatesExtension.php]: I think it'd be good to look for this on the `UpdatePackageInfoTask` like `$allowedTypes` above. If you could change to that, we should also add the private static to this extension with a short doc block explaining what it's for.

**Review by robbieaverill — CHANGES_REQUESTED** [review]

File: src/Extensions/CheckComposerUpdatesExtension.php (hunk 1 above)
Comment (Leapfrognz) [reply src/Extensions/CheckComposerUpdatesExtension.php]: Done

Comment (robbieaverill) [reply src/Extensions/CheckComposerUpdatesExtension.php]: Would you mind adding the static to this class too please?

```php
/**
 * Lists packages that will be ignored by the Composer update checker
 *
 * @config
 * @var string[]
 */
private static $ignored_packages = [];
```

Or something similar

Comment (NightJar) [reply src/Extensions/CheckComposerUpdatesExtension.php]: OK :)
Thanks for allowing contributors to make changes @Leapfrognz, I can keep this moving along.

**Review by robbieaverill — APPROVED** [review]

**Comment by rafaeldsousa** [comment]
@robbieaverill could we get this into version 2.0 and above ? Seems like it was never added there, but this is also something we could use on SS4.

**Pull request merged**